virtool/samples/api.py

```diff
@@ -855,20 +855,17 @@ async def upload_reads(req):
     if not await db.samples.find_one(sample_id):
         return not_found()
 
-    existing_reads = await virtool.samples.files.get_existing_reads(pg, sample_id)
-
-    if name in existing_reads:
-        return conflict("Reads file is already associated with this sample")
-
     try:
         size = await virtool.uploads.utils.naive_writer(req, reads_path, is_gzip_compressed)
     except OSError:
         return bad_request("File is not compressed")
     except asyncio.CancelledError:
         logger.debug(f"Reads file upload aborted for {sample_id}")
         return aiohttp.web.Response(status=499)
-
-    reads = await virtool.samples.files.create_reads_file(pg, size, name, name, sample_id, upload_id=upload)
+    try:
+        reads = await virtool.samples.files.create_reads_file(pg, size, name, name, sample_id, upload_id=upload)
+    except exc.IntegrityError:
+        return conflict("Reads file is already associated with this sample")
```

Comment:
Let's be explicit that it is the name that is the problem.
```suggestion
        return conflict("Reads file name is already uploaded for this sample")
```